src/main.cpp

```diff
@@ -16,7 +16,21 @@ int main(int argc, char** argv, char** envp) {
             Debug::quiet = true;
 
         else if (arg == "--config" || arg == "-c") {
+            if (i + 1 >= argc) {
+                Debug::log(NONE, "After " + arg + " you should provide a path to a config file.");
+                return 1;
+            }
+
+            if (!configPath.empty()) {
+                Debug::log(NONE, "Multiple config files are provided.");
+                return 1;
+            }
+
             configPath = argv[++i];
+            if (configPath[0] == '-') { // Should be fine, because of the null terminator
```

Comment:
comment is wrong, needs a check for configPath.empty()

Reply:
https://en.cppreference.com/w/cpp/string/basic_string/operator_at
For a `std::string` this will return null terminator if `pos == size()` (if string is empty size = 0, and pos = 0)

Reply:
oh, well, TIL. alright.